src/prefix-color-selector.ts

```diff
@@ -0,0 +1,77 @@
+import chalk from 'chalk';
+
+export class PrefixColorSelector {
+    lastColor: string;
+    autoColors: string[];
+
+    get ACCEPTABLE_CONSOLE_COLORS() {
+        // Colors picked randomly, can be amended if required
+        return (
+            [
+                chalk.cyan,
+                chalk.yellow,
+                chalk.magenta,
+                chalk.grey,
+                chalk.bgBlueBright,
+                chalk.bgMagenta,
+                chalk.magentaBright,
+                chalk.bgBlack,
+                chalk.bgWhite,
+                chalk.bgCyan,
+                chalk.bgGreen,
+                chalk.bgYellow,
+                chalk.bgRed,
+                chalk.bgGreenBright,
+                chalk.bgGrey,
+                chalk.blueBright,
+            ]
+                // Filter out duplicates
+                .filter((chalkColor, index, arr) => {
+                    return arr.indexOf(chalkColor) === index;
+                })
+                .map(chalkColor => chalkColor.bold.toString())
+        );
+    }
+
+    constructor(private readonly prefixColors?: string[], private readonly color?: boolean) {}
+
+    getNextColor(index?: number) {
```

Comment:
```suggestion
    getNextColor(index: number): string {
```

It's always being called with a value, so no need to make it optional

Reply:
Can we also add a jsdoc to this method?

Reply:
updated